sdk/ai/Azure.AI.Inference/tests/Utilities/ValidatingActivityListener.cs

```diff
@@ -0,0 +1,239 @@
+            }
+            Assert.AreNotEqual(expected.Name, "None");
+            Assert.Greater(expected.Timestamp, new DateTimeOffset(new DateTime(2024, 6, 1)));
+            ValidateEventTags(expected.Tags, actuals);
+        }
+
+        private static void ValidateChatMessageEvents(Activity activity, IList<ChatRequestMessage> messages)
+        {
+            Assert.NotNull(activity);
+            foreach (ChatRequestMessage message in messages)
+            {
+                ValidateEvent(
+                    activity: activity,
+                    name: $"gen_ai.{message.Role}.message",
+                    actuals: new Dictionary<string, object>()
+                    {
+                        { GenAiSystemKey, GenAiSystemValue},
+                        { GenAiEventContent, OpenTelemetryScope.GetContent(message)}
+                    }
+                );
+            }
+        }
+
+        private static void ValidateTag(Activity activity, string key, object value)
+        {
+            if (string.IsNullOrEmpty(value?.ToString()))
+            {
+                Assert.IsNull(activity.GetTagItem(key));
+            }
+            else
+            {
+                Assert.NotNull(activity.GetTagItem(key));
+                Assert.AreEqual(value.ToString(), activity.GetTagItem(key).ToString());
+            }
+        }
+
+        private static void ValidateFloatTag(Activity activity, string key, float? value)
+        {
+            if (!value.HasValue)
+            {
+                Assert.IsNull(activity.GetTagItem(key));
+            }
+            else
+            {
+                Assert.NotNull(activity.GetTagItem(key));
+                Assert.AreEqual(
+                    expected: float.Parse(value.ToString()),
+                    actual: float.Parse(activity.GetTagItem(key).ToString()),
+                    delta: 0.001
+                    );
+            }
+        }
+
+        private static void ValidateIntTag(Activity activity, string key, long? value)
+        {
+            if (!value.HasValue)
+            {
+                Assert.IsNull(activity.GetTagItem(key));
+            }
+            else
+            {
+                Assert.NotNull(activity.GetTagItem(key));
+                Assert.That(int.TryParse(activity.GetTagItem(key).ToString(), out int intActual));
+                Assert.AreEqual(value, intActual);
+            }
+        }
+
+        private static void ValidateEventTags(IEnumerable<KeyValuePair<string, object>> actuals, Dictionary<string, object> tags)
+        {
+            // Though technically we can add the same named tag multiple times, we are not doing it in
+            // our code.
+            Assert.AreEqual(tags.Count, actuals.Count());
+            foreach (KeyValuePair<string, object> actual in actuals)
+            {
+                Assert.That(tags.ContainsKey(actual.Key));
+                Assert.AreEqual(tags[actual.Key].ToString(), actual.Value.ToString());
```

Comment:
please validate without tostring - we need to make sure that types are right too